enshrine-esdt-safe/src/to_sovereign/create_tx.rs

```diff
@@ -130,7 +130,7 @@ pub trait CreateTxModule:
         refundable_payments: &ManagedVec<EsdtTokenPayment>,
     ) {
         for payment in refundable_payments {
-            self.send().direct_non_zero_esdt_payment(caller, &payment);
+            self.tx().to(caller).payment(&payment).transfer();
```

Comment:
You need a check if payment is non zero so you won't make transactions with 0 value. Or is the transfer function already checking this?

Reply:
added `require`